### ISSUE TITLE
TS-Compiler changes js-Filename depending on import-statement

### ISSUE THREAD

**Issue by YannickCAS** [Microsoft/TypeScript#5681]
Hello!

The Compiler changes the Output name of a TS-File depending on the first import-Statement.
In the example down below the file Dep.ts will get compiled into  dep.js and dep.map.js. If the order of the "test" and "test2" import statement changes it will get compiled into Dep.js and Dep.map.js.

Dep.ts should always compile to Dep.js.

Greetings

Yannick

class Dep{
    constructor(){

    }
}
export = Dep

----------------------------------------------------------------

import Dep = require("./dep")

export function test(){
    return new Dep();
}

----------------------------------------------------------------

import Dep = require("./Dep")

export function test2(){
    return new Dep();
}

----------------------------------------------------------------

import test = require("./test")
import test2 = require("./test2")

class CompilerStarter{
    constructor(){

    }
}
export = CompilerStarter

**Comment by vladima** [comment]
you can use `forceConsistentCasingInFileNames` compiler option to ensure that all references to some file file in program has the same casing. This is what I get on your code with this option enabled:
```
test.ts(1,22): error TS1149: File name 'C:/Sources/1/dep.ts' differs from already included file name 'C:/Sources/1/Dep.ts' only in casing
```

**Comment by benjamin-hg** [comment]
@vladima by closing this issue, do you really mean that the order in which imports appear in the code should affect the name casing of the output files?
I'd rather expect that the file name casing of the generated .js files matches exactly the casing of the respective .ts file!

**Comment by vladima** [comment]
currently we don't retrieve precise file name from the file system since it quite expensive both in perf (runtime overhead) and in development cost (need to maintain cross platform solution for all supported scenarios). Compiler computes list of files by doing depth first traversal of root files following imports and tripleslash references and picks whatever file name it found first (either in import or in the list of root files) assuming that on case insensitive systems they will refer to the same file. `forceConsistentCasingInFileNames` allows user to force the rule: 'all references to the same file should have the same casing'

**Comment by YannickCAS** [comment]
Thanks for the answer!
@vladima which version do I need to find this option?

**Comment by vladima** [comment]
this flag is already available in master and should be in TypeScript 1.8
